usr/iscsiadm.c

```diff
@@ -3013,8 +3013,78 @@ static int exec_fw_disc_op(discovery_rec_t *drec, struct list_head *ifaces,
 	return rc;
 }
 
+static int fill_in_default_fw_values(node_rec_t *rec, struct list_head *params)
+{
+	struct user_param *param;
+	int rc;
+
+	/* handle compatability issues for iface.transport_name */
+	if (!list_empty(params)) {
+		rc = verify_node_params(params, rec);
+		if (rc)
+			return rc;
+	}
+
+	/* Must init this so we can check if user overrode them */
+	rec->session.initial_login_retry_max = -1;
+	rec->conn[0].timeo.noop_out_interval = -1;
+	rec->conn[0].timeo.noop_out_timeout = -1;
+	rec->session.scan = -1;
+
+	list_for_each_entry(param, params, list) {
+		/*
+		 * user may not have passed in all params that were set by
+		 * ibft/iscsi_boot, so clear out values that might conflict
+		 * with user overrides
+		 */
+		if (!strcmp(param->name, IFACE_NETNAME)) {
+			/* overriding netname so MAC will be for old netdev */
+			memset(rec->iface.hwaddress, 0,
+				sizeof(rec->iface.hwaddress));
```

Comment:
This should be iface.netdev in the memset, or the comment is wrong, or I don't get it :)

Reply:
I think it's the latter. I'm not sure what is going on for all the iface fields.

Reply:
I remember this now :) I'm not sure if it's right for some cases now.

Can you make a helper function so it's not duplicated with iscsistart.

Reply:
> I remember this now :) I'm not sure if it's right for some cases now.
> 
> Can you make a helper function so it's not duplicated with iscsistart.

I want to change this comment after sleeping on it. It crept up in my dreams :(

I don't think we want this chunk and instead of the user passes them in fail. The iscsistart code is from a different time where we supported weird mix and match cases because offload boot and ibft/ppc boot were just coming out.

For the iscsiadm fw boot, we should start cleanly and not bring cruft over unless we need to. Since we currently only support booting from the interfaces that are presented in the iscsi_boot_sysfs interface (ibft/ppc fw/offload) then we should support it and not support/allow someone to override it.

So don't allow the user to specific those iface values.